src/index.ts

```diff
@@ -356,18 +336,61 @@ class LoopSDK {
     }
   }
 
+  logout() {
+    // only if there's a previous established connection
+    let hadConnected = false;
+
+    if (typeof window !== 'undefined') {
+      const raw = localStorage.getItem('loop_connect');
+      if (raw) {
+        try {
+          const parsed = JSON.parse(raw);
+          hadConnected = !!(parsed?.authToken || this.provider);
+        } catch {
+          // ignore parse errors
+        }
+      }
+
+      localStorage.removeItem('loop_connect');
+    }
+
+    this.ticketId = null;
+    this.provider = null;
+    this.connection?.ws?.close();
+    this.hideQrCode();
+  }
+
   private requireProvider(): Provider {
     if (!this.provider) {
       throw new Error('SDK not connected. Call connect() and wait for acceptance first.');
     }
     return this.provider;
   }
 
+  private async ensureSessionValidOrLogout(): Promise<void> {
+    if (!this.connection || !this.provider) return;
+
+    // verify the stored auth token
+    const authToken = (this.provider as any)?.auth_token as string | undefined;
+    if (!authToken) return;
+
+    try {
+      await this.connection.verifySession(authToken);
```

Comment:
in verifySession only logout if we got 401/403. If we get 5xx(500 error or 502-504 timeout we should not logout.